### PR TITLE
feat(faq): add `stdout`/`stderr` performance

### LINKED ISSUE #274: Add stdout / stderr performance to FAQ
https://ratatui.rs/faq/#should-i-use-stdout-or-stderr

See https://github.com/ratatui-org/ratatui/pull/583#discussion_r1425436137

[Valentin271](https://github.com/Valentin271) [Dec 13, 2023](https://github.com/ratatui-org/ratatui/pull/583#discussion_r1425436137)

Oh that's interesting, stdout is also faster on my machine

    from ~1.60 to ~2 fps in fullscreen
    from ~15 to ~90 fps in 80x24!

If this is consistent, we might want to keep that in mind. We currently use stderr in the template. And we should mention that in the [FAQ](https://ratatui.rs/faq/#should-i-use-stdout-or-stderr)

### PR DESCRIPTION
**Pull request by Valentin271**
Fixes #274

### CODE REVIEW

**Review by kdheepak — APPROVED** [review]

File: src/content/docs/faq.md [hunk 1]
```diff
@@ -413,3 +413,17 @@ With `stderr`:
...
+easily make `stderr` buffered too by wrapping it in a [`BufWriter`] like so:
+
+```rust
+let mut terminal = Terminal::new(CrosstermBackend::new(std::io::stderr()))?;
+```
+
+Our recommendation is to use `stdout`. If you really need `stderr`, then accept the performance
+loss (which is unnoticeable in most applications) or make it buffered.
+
+If you want to know more, we recommend reading [this excellent article by orhun](https://blog.orhun.dev/stdout-vs-stderr/).
```
Comment (orhun) [inline src/content/docs/faq.md]:
```suggestion
If you want to know more, we recommend reading [this excellent article by @orhun](https://blog.orhun.dev/stdout-vs-stderr/).
```

File: src/content/docs/faq.md [hunk 2]
```diff
@@ -413,3 +413,17 @@ With `stderr`:
 
 - No special setup necessary in order to run in a pipe command
 - Unconventional and that might subvert users expectations
+
+Out of the box, `stdout` will be faster than `stderr` because it is buffered. However you can very
+easily make `stderr` buffered too by wrapping it in a [`BufWriter`] like so:
+
+```rust
+let mut terminal = Terminal::new(CrosstermBackend::new(std::io::stderr()))?;
```
Comment (orhun) [inline src/content/docs/faq.md]:
```suggestion
let mut terminal = Terminal::new(CrosstermBackend::new(BufWriter::new(std::io::stderr())))?;
```

**Review by orhun — COMMENTED** [review]
💖💖💖

File: src/content/docs/faq.md (hunk 2 above)
Comment (Valentin271) [reply src/content/docs/faq.md]: Right I forgot the most important :sweat_smile: , fixed

**Review by orhun — APPROVED** [review]

**Pull request merged**